src/lib.rs

```diff
@@ -0,0 +1,94 @@
+use aes::Aes128;
+use cmac::{Cmac, Mac, NewMac};
+use array2d::Array2D;
+
+//use generic_array::{ArrayLength, GenericArray};
+
+pub fn test_mac() -> () {
+	// Create `Mac` trait implementation, namely CMAC-AES128
+	let mut mac = Cmac::<Aes128>::new_from_slice(b"very secret key.").unwrap();
+	mac.update(b"input message");
+
+	// `result` has type `Output` which is a thin wrapper around array of
+	// bytes for providing constant time equality check
+	let result = mac.finalize();
+	// To get underlying array use the `into_bytes` method, but be careful,
+	// since incorrect use of the tag value may permit timing attacks which
+	// defeat the security provided by the `Output` wrapper
+	let tag_bytes = result.into_bytes();
+	
+	let mut mac2 = Cmac::<Aes128>::new_from_slice(b"very secret key.").unwrap();
+
+	mac2.update(b"input message");
+
+	// `verify` will return `Ok(())` if tag is correct, `Err(MacError)` otherwise
+	mac2.verify(&tag_bytes).unwrap()
+}
+
+pub fn create_mac(fcnt: u32) -> &'static [u16] {
+	unsafe {
+	let mut mac = Cmac::<Aes128>::new_from_slice(b"very secret key.").unwrap();

```

Comment:
i would suggest passing a reference to a mac and cloning it here to make it clear to the reader that it's the same "session"